File: app/src/main/java/com/example/weizhenbin/wangebug/tools/preferences/PreferencesTool.java
package com.example.weizhenbin.wangebug.tools.preferences;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.annotation.Nullable;

import com.example.weizhenbin.wangebug.base.App;

import java.util.Set;

/**
 * Created by weizhenbin on 2018/9/18.
 */

public final class PreferencesTool {



    public static void putString(String key, @Nullable String value){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(key,value);
        editor.apply();
    }

    public static void putStringSet(String key, @Nullable Set<String> values){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        SharedPreferences.Editor editor = preferences.edit();
        editor.putStringSet(key,values);
        editor.apply();
    }
    public static void putInt(String key, int value){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        SharedPreferences.Editor editor = preferences.edit();
        editor.putInt(key,value);
        editor.apply();
    }
    public static void putLong(String key, long value){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        SharedPreferences.Editor editor = preferences.edit();
        editor.putLong(key,value);
        editor.apply();
    }
    public static void putFloat(String key, float value){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        SharedPreferences.Editor editor = preferences.edit();
        editor.putFloat(key,value);
        editor.apply();
    }
    public static void putBoolean(String key, boolean value){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean(key,value);
        editor.apply();
    }

    public static boolean getBoolean(String key, boolean defValue){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        return preferences.getBoolean(key,defValue);
    }

    public static long getLong(String key, long defValue){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        return preferences.getLong(key,defValue);
    }
    public static int getInt(String key, int defValue){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        return preferences.getInt(key,defValue);
    }
    public static float getFloat(String key, float defValue){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        return preferences.getFloat(key,defValue);
    }
    public static String getString(String key, String defValue){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        return preferences.getString(key,defValue);
    }
    public static Set<String> getStringSet(String key, @Nullable Set<String> defValues){
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(App.Companion.getApp().getApplicationContext());
        return preferences.getStringSet(key,defValues);
    }
}

File: app/src/main/java/com/example/weizhenbin/wangebug/tools/preferences/PreferencesTool.kt
package com.example.weizhenbin.wangebug.tools.preferences

import android.preference.PreferenceManager
import com.example.weizhenbin.wangebug.base.App

/**
 * Created by weizhenbin on 2018/9/18.
 */

object PreferencesTool {


    fun putString(key: String, value: String?) {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        val editor = preferences.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun putStringSet(key: String, values: Set<String>?) {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        val editor = preferences.edit()
        editor.putStringSet(key, values)
        editor.apply()
    }

    fun putInt(key: String, value: Int) {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        val editor = preferences.edit()
        editor.putInt(key, value)
        editor.apply()
    }

    fun putLong(key: String, value: Long) {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        val editor = preferences.edit()
        editor.putLong(key, value)
        editor.apply()
    }

    fun putFloat(key: String, value: Float) {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        val editor = preferences.edit()
        editor.putFloat(key, value)
        editor.apply()
    }

    fun putBoolean(key: String, value: Boolean) {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        val editor = preferences.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun getBoolean(key: String, defValue: Boolean): Boolean {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        return preferences.getBoolean(key, defValue)
    }

    fun getLong(key: String, defValue: Long): Long {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        return preferences.getLong(key, defValue)
    }

    fun getInt(key: String, defValue: Int): Int {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        return preferences.getInt(key, defValue)
    }

    fun getFloat(key: String, defValue: Float): Float {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        return preferences.getFloat(key, defValue)
    }

    fun getString(key: String, defValue: String): String? {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        return preferences.getString(key, defValue)
    }

    fun getStringSet(key: String, defValues: Set<String>?): Set<String>? {
        val preferences = PreferenceManager.getDefaultSharedPreferences(App.app.applicationContext)
        return preferences.getStringSet(key, defValues)
    }
}
